### PR TITLE
Carpool UI

### LINKED ISSUE #31: Carpool routing UI
### Button
- [x] replace the icon by a car showing two heads
- [x] the button should be displayed in the transit options

### PR DESCRIPTION
**Pull request by Beck-berry**
Fixes #31 
## Proposed Changes
  - created new mfbw configuration
  - added carpooling as transport mode
  - added icon and color for carpooling

### CODE REVIEW

**Comment by leonardehrenfried** [comment]
Could you please add the carpooling relevant config to the hb config file? I would like to get this on the staging server (https://mih.mitfahren-bw.de/) so we can start integrating already - even if this means that we cannot find any routes.

**Comment by leonardehrenfried** [comment]
Screenshot from 2020-02-13 19-12-25

I can also see the icon as a sub-point of Park & Ride. Is this where you expect it?

**Comment by Beck-berry** [comment]
It is given as a transport mode so basically it should be available when all the transport modes are - so with park&ride too.
Do you think it should be used/added otherwise?

**Comment by leonardehrenfried** [comment]
No, if that is the intention then it's fine. 

I was just wondering if we should make such an innovative feature a little more prominent, but perhaps it is wise to start a little more hidden.

**Comment by leonardehrenfried** [comment]
@hbruch this looks good to me. Please review and merge.

**Comment by leonardehrenfried** [comment]
One last question: do we still need the extra config file?

File: app/translations.js [hunk 1]
```diff
@@ -314,6 +314,7 @@ const translations = {
     car: 'Auto',
     'car-distance-duration': 'Auto {distance} ({duration})',
     car_park: 'Park & Ride',
+    carpool: 'Carpool', // TODO add German translation
```
Comment (hbruch) [inline app/translations.js]: "Fahrgemeinschaft" or "Mitfahren", I prefer "Fahrgemeinschaft"

File: sass/themes/default/_theme.scss [hunk 2]
```diff
@@ -63,6 +63,7 @@ $tram-color: $livi-tram-green;
 $metro-color: $livi-subway-orange;
 $rail-color: $livi-rail-purple;
 $ferry-color: $livi-ferry-blue;
+$carpool-color: #800000;
```
Comment (hbruch) [inline sass/themes/default/_theme.scss]: I'd prefer to use the Herrenberg green (which is close to the Fahrgemeinschaft green) as color for carpools. But then we should change the bus symbol background to either purple as in https://de.wikipedia.org/wiki/Datei:BUS-Logo.svg  or better the VVS red https://de.wikipedia.org/wiki/Datei:Bus-Logo_VVS.svg.
For the train, we could use https://de.wikipedia.org/wiki/Datei:S-Bahn-Logo.svg
And metro should be activated as well (some people want to got to Stuttgart where it exists), best would be the U-Bahn blue https://de.wikipedia.org/wiki/Datei:U-Bahnlogo_München.png

**Comment by hbruch** [comment]
Before merging, we should complete the whole epic. Or we toggle the feature via a config param, so it does not slip into production before completed. Then it would be a parameter to en/disable the feature depending on environment.

**Comment by leonardehrenfried** [comment]
> Before merging, we should complete the whole epic. Or we toggle the feature via a config param, so it does not slip into production before completed. Then it would be a parameter to en/disable the feature depending on environment.

I'm strongly in favour of some sort of feature toggle rather than holding back the merge, as this IMO leads to way more problems. In any case, would it be so terrible to have this feature in production? It's quite hidden and if a user clicks it, nothing will happen.

**Comment by Beck-berry** [comment]
### Colors
I set the carpool color in the default theme to the hb green, and changed the color of the other transport modes as requested.

Screenshot_2020-02-14_10-21-44 Screenshot_2020-02-14_11-14-21

Are these ok, @hbruch ?

I had to create a hb-specific svg file in order to modify the icons. (Last time I made changes to the svg-sprite.default.svg which was a bad idea - we only want to modify them in our config, so I reverted those changes, this will show up as modification.)

**Comment by hbruch** [comment]
Colors are good. I just realize that the M for Metro should be an U for "U-Bahn" here. Could you change it so It looks like https://de.wikipedia.org/wiki/Datei:U-Bahnlogo_München.png?

**Comment by hbruch** [comment]
And we should replace "Zug" by "S-Bahn / Zug"

**Comment by leonardehrenfried** [comment]
I tried it out and it looks good: it only displays the button when the env variable is set and the colours look nice.

There is one problem though, when you select a carpool route, the UI breaks. This is a backend problem though as the mode of some legs is null. This should be fixed in the latest OTP version.

Screenshot from 2020-02-14 16-10-21

@hbruch I recommend that we merge this in and I disable the button again. Then I can enable it again when the backend is ready.

File: app/component/CarLeg.js [hunk 3]
```diff
@@ -42,10 +54,13 @@ function CarLeg(props, context) {
...
         <div className="itinerary-leg-action">
           <FormattedMessage
-            id="car-distance-duration"
+            id={CarLeg.getTranslationKey(props.leg.mode)}
             values={{ distance, duration }}
             defaultMessage="Drive {distance} ({duration})}"
           />
+          {CarLeg.createBookButton(props.leg)}
+          <Icon img={carpoolAgencyIcon[0]} className="carpool-agency-logo" />
+          <Icon img={carpoolAgencyIcon[1]} className="carpool-agency-logo" />
```
Comment (hbruch) [inline app/component/CarLeg.js]: Mifaz has only one icon. We should not generate an empty img tag here in this case.

**Pull request merged**